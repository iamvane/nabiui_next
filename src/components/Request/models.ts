import { number } from "../StudioInstructor/constants"

export interface ApplicationPayload {
  requestId: number;
  userId: string;
  accept: Boolean;
}

export interface StudentType {
  name: string;
  age: number;
}

export interface RequestAvailabilityType {
  day: string;
  timeframe: string;
}

export interface RequestType {
  id?: number;
  createdAt?: string;
  elapsedTime?: string;
  displayName?: string;
  instrument?: string;
  lessonDuration?: string;
  requestTitle?: string;
  requestMessage?: string;
  placeForLessons?: string;
  skillLevel?: string;
  role?: string;
  applicationsReceived?: number;
  applied?: boolean;
  studentDetails?: StudentType[];
  avatar?: string;
  location?: string;
  status?: string;
  availability: RequestAvailabilityType[];
  studentId?: number;
  language?: string;
  timezone?: string;
  gender?: string;
}
