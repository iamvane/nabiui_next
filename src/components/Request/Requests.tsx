import * as React from 'react';

import { Request } from '../../redux/models/RequestModel';
import { UserType } from '../../redux/models/UserModel';
import RequestCard from './RequestCard';

interface Props {
  requests: Request[];
  isRequesting: boolean;
  isLoggedIn: boolean;
  toggleRegisterModal: () => void;
}

const Requests: React.StatelessComponent<Props> = props => {
  return (
    <div>
      {props.requests && props.requests.map((request, i) => (
        <RequestCard key={i} request={request} isLoggedIn={props.isLoggedIn} toggleRegisterModal={props.toggleRegisterModal} />
      ))}
    </div>
  );
};

export default Requests;
