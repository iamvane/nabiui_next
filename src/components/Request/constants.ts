export namespace NewRequestComponent {
  export const pageTitle = 'New Student';
  export const title = 'Trial {instrument} Lesson';
  export const instrumentPlaceholder = '{instrument}';
  export const date = '{dateRepalce} @ {timeReplace}';
  export const datePlaceholder = '{dateRepalce}';
  export const timePlaceholder = '{timeReplace}';
  export const studentDetails = '{studentName}, {studentAge} years old, {skillLevel}';
  export const studentNamePlaceholder = '{studentName}';
  export const agePlaceholder = '{studentAge}';
  export const skillLevelPlaceholder = '{skillLevel}';
  export const lessonDuration = '30 mins';
  export const acceptButton = 'Accept';
  export const passButton = 'Pass';
  export const goToStudioButton = 'Go To Studio';
  export const errorMessage = 'There was an error with your request. Please refresh the page and try again.';
  export const responseMessage = 'Thanks for your response.';

  export const weekdaysLabels = {
    'mon': 'Monday',
    'tue': 'Tuesday',
    'wed': 'Wednesday',
    'thu': 'Thursday',
    'fri': 'Friday',
    'sat': 'Saturday',
    'sun': 'Sunday'
  }
  
  export const timeframeLabels = {
    'early-morning': 'early morning (8am-10am)',
    'late-morning': 'late morning (10am-12pm)',
    'early-afternoon': 'early afternoon (12pm-3pm)',
    'late-afternoon': 'late afternoon (3pm -6pm)',
    'evening': 'evening (6pm-9pm)'
  }

}
