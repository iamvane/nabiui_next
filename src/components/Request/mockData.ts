export const requestsMockData = [
  {
    id: 0,
    applied: false,
    user: {
      id: '0',
      email: 'rox@gmail.com',
      role: 'student',
      firstName: 'Roxana',
      lastName: 'Gallardo',
      displayName: 'Roxana G.',
      distance: '0.5 mi away',
      birthday: '1996-07-12T04:00:00.000Z'
    },
    instrument: 'piano',
    placeForLessons: 'studio',
    lessonDuration: '45',
    location: 'Boston, Ma',
    applications: 17,
    createdAt: '2019-08-05T23:01:25.000Z',
    title: 'Fun and experience piano instructor',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Roxana',
        age: 18,
        skillLevel: 'beginner'
      },
    ]
  },
  {
    id: 1,
    applied: true,
    application: {
      createdAt: 'Mon Aug 22, 2019 at 2:00pm EDT',
      rate: 50,
      message: 'Hello Sam, I would love to work with your family and typesetting industria  Ipsum is simply dummy.'
    },
    user: {
      id: '1',
      email: 'sam@gmail.com',
      role: 'parent',
      firstName: 'Sam',
      lastName: 'Flood',
      displayName: 'Sam F.',
      distance: '0.5 mi away'
    },
    instrument: 'piano',
    placeForLessons: 'home',
    lessonDuration: '45',
    createdAt: '2019-08-05T20:19:25.000Z',
    location: 'Boston, Ma',
    applications: 17,
    title: 'Piano teacher needed for my family',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Zoe',
        age: 6,
        skillLevel: 'beginner'
      },
      {
        name: 'Esme',
        age: 9,
        skillLevel: 'beginner'
      }
    ]
  },
  {
    id: 2,
    applied: false,
    user: {
      id: '2',
      email: 'victor@gmail.com',
      role: 'parent',
      firstName: 'Victor',
      lastName: 'Charles',
      displayName: 'Victor C.',
      distance: '1 mi away'
    },
    instrument: 'piano',
    placeForLessons: 'home',
    lessonDuration: '45',
    createdAt: '2019-08-05T15:19:25.000Z',
    location: 'Boston, Ma',
    applications: 17,
    title: 'Instructor for one child in Boston',
    message: 'Lorem Ipsum is simply dummy text of the printing and typesetting industria',
    students: [
      {
        name: 'Ethan',
        age: 11,
        skillLevel: 'beginner'
      },
    ]
  }
];
