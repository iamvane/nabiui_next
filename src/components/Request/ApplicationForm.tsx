import * as React from 'react';
import NumberFormat from 'react-number-format';
import {
  Button,
  Grid,
  TextField,
  Typography
} from '@material-ui/core';

import SectionTitle from '../common/SectionTitle';
import { RequestViewComponent } from './constants';

interface Props {
  // TODO: set to RequestType on api integration
  values: any;
  request: any;
  handleChange: (event: React.FormEvent<{}>) => void;
  handleSubmit: (event: React.FormEvent<{}>) => void;
}

function NumberFormatCustom(props: any) {
  const { inputRef, onChange, name,  ...other } = props;

  return (
    <NumberFormat
      {...other}
      ref={inputRef}
      onValueChange={values => {
        onChange({
          currentTarget: {
            value: values.value,
            name: name
          },
        });
      }}
      decimalScale={2}
      fixedDecimalScale={true}
      thousandSeparator={true}
    />
  );
}

const ApplicationForm: React.StatelessComponent<Props> = props => {
  return (
    <React.Fragment>
      <SectionTitle text={RequestViewComponent.sendApplication} />
      <Grid className="nabi-text-center nabi-margin-bottom-small">
        <Typography
          className="nabi-text-mediumbold nabi-margin-top-small nabi-text-uppercase"
        >
          {RequestViewComponent.Labels.LessonRate}
        </Typography>
        <Typography
          className="nabi-margin-bottom-xsmall"
        >
        {RequestViewComponent.lessonDuration.replace(
          RequestViewComponent.lessonDurationPlaceHolder,
          props.request.lessonDuration
        )}
        </Typography>
        <TextField
          className="nabi-rates-field"
          placeholder="$0.00"
          value={(props as any)[props.values.lessonRate]}
          onChange={props.handleChange}
          id={RequestViewComponent.Ids.LessonRate}
          name={RequestViewComponent.FieldNames.LessonRate}
          InputProps={{
            inputComponent: NumberFormatCustom,
          }}
        />
      </Grid>
      <Typography
        className="nabi-text-mediumbold nabi-text-center nabi-margin-bottom-xsmall nabi-text-uppercase"
      >
        {RequestViewComponent.sendMessageTo.replace(
          RequestViewComponent.userPlaceholder,
          props.request.user && props.request.user.firstName
        )}
      </Typography>
      <TextField
        margin="normal"
        className="nabi-margin-remove"
        required={true}
        multiline={true}
        onChange={props.handleChange}
        value={props.values.message}
        id={RequestViewComponent.Ids.Message}
        name={RequestViewComponent.FieldNames.Message}
        fullWidth={true}
        rows={6}
      />
      <Grid item={true} xs={12} className="nabi-text-center">
        <Button
          color="primary"
          onClick={props.handleSubmit}
          variant="contained"
          className="nabi-margin-top-small"
        >
          {RequestViewComponent.sendApplication}
        </Button>
      </Grid>
    </React.Fragment>
  );
};

export default ApplicationForm;
