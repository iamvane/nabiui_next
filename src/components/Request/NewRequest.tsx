import * as React from 'react';
import Link from 'next/link';
import {
  Action,
  Dispatch
} from 'redux';
import { connect } from 'react-redux';
import { useRouter } from 'next/router';

import {
  Button,
  Grid,
  Typography,
  CircularProgress
} from '@material-ui/core';
import DateRangeIcon from '@material-ui/icons/DateRange';
import Schedule from '@material-ui/icons/Schedule';
import Face from '@material-ui/icons/Face';
import MusicNoteIcon from '@material-ui/icons/MusicNote';
import LocationOnOutlinedIcon from '@material-ui/icons/LocationOnOutlined';
import PublicOutlinedIcon from '@material-ui/icons/PublicOutlined';

import { StoreState } from '../../redux/reducers/store';
import { submitApplication } from '../../redux/actions/InstructorActions';
import { fetchRequest } from '../../redux/actions/RequestActions';
import { instrumentDisplay } from "../../utils/displayInstrument";
import PageTitle from '../common/PageTitle';
import { Routes } from '../common/constants/Routes';
import { ApplicationPayload, RequestType } from "./models";
import { NewRequestComponent } from './constants'

interface OwnProps {
}

interface DispatchProps {
  fetchRequest: (id: number) => void;
  respond: (accept: ApplicationPayload) => void;
}

interface StateProps {
  request: RequestType;
  isFetchingRequest: boolean;
  isResponding: boolean;
  respondMessage: string
  respondError: string;
}

interface Props extends
  DispatchProps,
  StateProps { }




const NewRequest = (props: Props) => {

  const router = useRouter();
  const id = Number(router.query.id);

  React.useEffect(() => {
    const fetchData = async () => {
      if (id) {
        await props.fetchRequest(id);
      }
    };
    fetchData();
    /* tslint:disable */
  }, []);

  const handleSubmit = async (accept: Boolean) => {
    await props.respond({
      requestId: id,
      userId: String(id),
      accept
    })
  }

  const displayAvailability = () => {
    let availability = [];
    props.request.availability.map(item =>
      availability.push(NewRequestComponent.weekdaysLabels[item.day] + ' ' + NewRequestComponent.timeframeLabels[item.timeframe])
    )
    return availability.join(', ').replace(/, ([^,]*)$/, ' and $1');
  }

  return (
    <div className="nabi-container nabi-margin-bottom-medium">
      <PageTitle pageTitle={NewRequestComponent.pageTitle} />
      <Grid
        item={true}
        xs={12}
        md={8} className="nabi-section nabi-background-white nabi-margin-center"
      >
        {props.isFetchingRequest || props.isResponding ?
          <div className="nabi-text-center"><CircularProgress /></div> :
          props.respondError ?
            <Typography>{props.respondError}</Typography> :
            props.request.status === 'CLOSED' ? (
              <div className="nabi-text-center">
                <Typography>{NewRequestComponent.closedRequestMessage}</Typography>
                <Link href={Routes.InstructorStudio}>
                  <Button color="primary" variant="contained" className="nabi-margin-top-small">
                    {NewRequestComponent.goToStudioButton}
                  </Button>
                </Link>
              </div>
            ) :
            props.respondMessage ?
              <div className="nabi-text-center">
                <Typography>{NewRequestComponent.responseMessage}</Typography>
                <Link href={Routes.InstructorStudio}>
                  <Button color="primary" variant="contained" className="nabi-margin-top-small">
                    {NewRequestComponent.goToStudioButton}
                  </Button>
                </Link>
              </div> :
              <>
                <p className="nabi-color-nabi nabi-text-center nabi-jennasue-title nabi-margin-bottom-small nabi-margin-top-xsmall">
                  {NewRequestComponent.title.replace(
                    NewRequestComponent.instrumentPlaceholder,
                    instrumentDisplay(props.request.instrument)
                  )}
                </p>
                <div>
                  <DateRangeIcon className="text-aligned-icon" color="primary" />
                  <Typography className="nabi-display-inline nabi-margin-left-xsmall">
                    {displayAvailability()}
                  </Typography>
                </div>
                <div className="nabi-display-flex nabi-margin-top-xsmall">
                  <LocationOnOutlinedIcon color="primary" />
                  <Typography color="primary" className="nabi-margin-left-xsmall">
                    {props.request.timezone}
                  </Typography>
                </div>
                <div>
                  <Schedule className="text-aligned-icon" color="primary" />
                  <Typography className="nabi-display-inline nabi-margin-left-xsmall">
                    {NewRequestComponent.lessonDuration}
                  </Typography>
                </div>
                <div>
                  <MusicNoteIcon className="text-aligned-icon" color="primary" />
                  <Typography className="nabi-display-inline nabi-margin-left-xsmall">
                    {instrumentDisplay(props.request.instrument)}
                  </Typography>
                </div>
                <div>
                  <Face className="text-aligned-icon" color="primary" />
                  {props.request.studentDetails &&
                    <Typography className="nabi-display-inline nabi-margin-left-xsmall">
                      {NewRequestComponent.studentDetails.replace(
                        NewRequestComponent.studentNamePlaceholder,
                        props.request.studentDetails[0].name
                      ).replace(
                        NewRequestComponent.agePlaceholder,
                        String(props.request.studentDetails[0].age)
                      ).replace(
                        NewRequestComponent.skillLevelPlaceholder,
                        props.request.skillLevel
                      )}
                    </Typography>
                  }
                </div>
                <div className="nabi-display-flex nabi-margin-top-xsmall">
                  <PublicOutlinedIcon color="primary" />
                  <Typography color="primary" className="nabi-margin-left-xsmall">
                    {props.request.language}
                  </Typography>
                </div>
                <div className="nabi-text-right nabi-margin-top-large">
                  <Button
                    color="default"
                    className="nabi-margin-right-xsmall"
                    onClick={() => handleSubmit(false)}
                  >
                    {NewRequestComponent.passButton}
                  </Button>
                  <Button
                    color="primary"
                    variant="contained"
                    onClick={() => handleSubmit(true)}
                  >
                    {NewRequestComponent.acceptButton}
                  </Button>
                </div>
              </>}
      </Grid>
    </div>
  )
}

const mapStateToProps = (state: StoreState, _ownProps: OwnProps): StateProps => {
  const {
    isRequesting: isResponding,
    message: respondMessage,
    error: respondError,
  } = state.instructor.actions.submitApplication

  return {
    request: state.requests.request,
    isFetchingRequest: state.requests.actions.fetchRequest.isRequesting,
    isResponding,
    respondMessage,
    respondError
  };
};

const mapDispatchToProps = (
  dispatch: Dispatch<Action>
): DispatchProps => ({
  fetchRequest: (id: number) => dispatch(fetchRequest(id)),
  respond: (applciation: ApplicationPayload) => dispatch(submitApplication(applciation))
});

export default connect(mapStateToProps, mapDispatchToProps)(NewRequest);
