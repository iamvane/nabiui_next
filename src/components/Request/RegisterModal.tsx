import * as React from 'react';
import Link from 'next/link';
import {
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle
} from '@material-ui/core';

import { Routes } from '../common/constants/Routes';

interface Props {
  isOpen: boolean;
  handleClose: () => void;
}

const RegisterModal: React.StatelessComponent <Props> = props => {
  return (
    <div>
      <Dialog
        open={props.isOpen}
        onClose={props.handleClose}
        aria-labelledby="form-dialog-title"
        scroll="body"
      >
        <DialogTitle id="form-dialog-title nabi-text-uppercase">REGISTER OR LOGIN TO APPLY</DialogTitle>

        <DialogContent>
          <DialogContentText>
            Sign up today and start teaching! Login if you have an account.
          </DialogContentText>
        </DialogContent>

        <DialogActions>
          <Link href={Routes.RegistrationInstructor}>
            <a>
              <Button variant="contained" color="primary">Register</Button>
            </a>
          </Link>
          <Link href={Routes.Login}>
            <a>
              <Button variant="contained" color="secondary" onClick={props.handleClose}>
                Login
              </Button>
            </a>
          </Link>
        </DialogActions>
      </Dialog>
    </div>
  );
};

export default RegisterModal;
