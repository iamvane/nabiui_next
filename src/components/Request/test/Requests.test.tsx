import * as React from 'react';
import { shallow } from 'enzyme';

import Requests from 'components/Request/Requests';
import { RequestType } from 'components/Request/models';

const requests: RequestType[] = [
  {
    instrument: 'Piano',
    placeForLessons: 'studio',
    lessonDuration: '45',
    location: 'Boston, Ma',
    applications: 17,
    createdAt: 'Sun Apr 21 2019 18:51:56',
    requestTitle: 'fun and experience piano instructor',
    requestMessage: 'Lorem Ipsum is simply dummy text of the printing',
    students: [
      {
        name: 'Peter',
        age: 18,
        skillLevel: 'Beginner'
      },
      {
        name: 'Sayvent',
        age: 15,
        skillLevel: 'Beginner'
      }
    ]
  }
];

describe('Requests', () => {
  const  wrapper = shallow(
    <Requests requests={requests} />
  );

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
