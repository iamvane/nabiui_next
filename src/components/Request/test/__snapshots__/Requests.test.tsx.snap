// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Requests Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Requests
    requests={
      Array [
        Object {
          "applications": 17,
          "createdAt": "Sun Apr 21 2019 18:51:56",
          "instrument": "Piano",
          "lessonDuration": "45",
          "location": "Boston, Ma",
          "placeForLessons": "studio",
          "requestMessage": "Lorem Ipsum is simply dummy text of the printing",
          "requestTitle": "fun and experience piano instructor",
          "students": Array [
            Object {
              "age": 18,
              "name": "Peter",
              "skillLevel": "Beginner",
            },
            Object {
              "age": 15,
              "name": "Sayvent",
              "skillLevel": "Beginner",
            },
          ],
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <RequestCard
          request={
            Object {
              "applications": 17,
              "createdAt": "Sun Apr 21 2019 18:51:56",
              "instrument": "Piano",
              "lessonDuration": "45",
              "location": "Boston, Ma",
              "placeForLessons": "studio",
              "requestMessage": "Lorem Ipsum is simply dummy text of the printing",
              "requestTitle": "fun and experience piano instructor",
              "students": Array [
                Object {
                  "age": 18,
                  "name": "Peter",
                  "skillLevel": "Beginner",
                },
                Object {
                  "age": 15,
                  "name": "Sayvent",
                  "skillLevel": "Beginner",
                },
              ],
            }
          }
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "request": Object {
            "applications": 17,
            "createdAt": "Sun Apr 21 2019 18:51:56",
            "instrument": "Piano",
            "lessonDuration": "45",
            "location": "Boston, Ma",
            "placeForLessons": "studio",
            "requestMessage": "Lorem Ipsum is simply dummy text of the printing",
            "requestTitle": "fun and experience piano instructor",
            "students": Array [
              Object {
                "age": 18,
                "name": "Peter",
                "skillLevel": "Beginner",
              },
              Object {
                "age": 15,
                "name": "Sayvent",
                "skillLevel": "Beginner",
              },
            ],
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <RequestCard
            request={
              Object {
                "applications": 17,
                "createdAt": "Sun Apr 21 2019 18:51:56",
                "instrument": "Piano",
                "lessonDuration": "45",
                "location": "Boston, Ma",
                "placeForLessons": "studio",
                "requestMessage": "Lorem Ipsum is simply dummy text of the printing",
                "requestTitle": "fun and experience piano instructor",
                "students": Array [
                  Object {
                    "age": 18,
                    "name": "Peter",
                    "skillLevel": "Beginner",
                  },
                  Object {
                    "age": 15,
                    "name": "Sayvent",
                    "skillLevel": "Beginner",
                  },
                ],
              }
            }
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "request": Object {
              "applications": 17,
              "createdAt": "Sun Apr 21 2019 18:51:56",
              "instrument": "Piano",
              "lessonDuration": "45",
              "location": "Boston, Ma",
              "placeForLessons": "studio",
              "requestMessage": "Lorem Ipsum is simply dummy text of the printing",
              "requestTitle": "fun and experience piano instructor",
              "students": Array [
                Object {
                  "age": 18,
                  "name": "Peter",
                  "skillLevel": "Beginner",
                },
                Object {
                  "age": 15,
                  "name": "Sayvent",
                  "skillLevel": "Beginner",
                },
              ],
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
